cmd/application_update.go

```diff
@@ -34,73 +33,12 @@ var applicationUpdateCmd = &cobra.Command{
 			panic("unreachable") // staticcheck false positive: https://staticcheck.io/docs/checks#SA5011
 		}
 
-		applications, _, err := client.ApplicationsAPI.ListApplication(context.Background(), envId).Execute()
-
-		if err != nil {
-			utils.PrintlnError(err)
-			os.Exit(1)
-			panic("unreachable") // staticcheck false positive: https://staticcheck.io/docs/checks#SA5011
-		}
-
-		application := utils.FindByApplicationName(applications.GetResults(), applicationName)
-
-		if application == nil {
-			utils.PrintlnError(fmt.Errorf("application %s not found", applicationName))
-			utils.PrintlnInfo("You can list all applications with: qovery application list")
-			os.Exit(1)
-			panic("unreachable") // staticcheck false positive: https://staticcheck.io/docs/checks#SA5011
-		}
-
-		var storage []qovery.ServiceStorageRequestStorageInner
-		for _, s := range application.Storage {
-			storage = append(storage, qovery.ServiceStorageRequestStorageInner{
-				Id:         &s.Id,
-				Type:       s.Type,
-				Size:       s.Size,
-				MountPoint: s.MountPoint,
-			})
-		}
-
-		req := qovery.ApplicationEditRequest{
-			Storage:     storage,
-			Name:        &application.Name,
-			Description: application.Description,
-			GitRepository: &qovery.ApplicationGitRepositoryRequest{
-				Branch:     application.GitRepository.Branch,
-				GitTokenId: application.GitRepository.GitTokenId,
-				RootPath:   application.GitRepository.RootPath,
-				Url:        application.GitRepository.Url,
-			},
-			BuildMode:           application.BuildMode,
-			DockerfilePath:      application.DockerfilePath,
-			BuildpackLanguage:   application.BuildpackLanguage,
-			Cpu:                 application.Cpu,
-			Memory:              application.Memory,
-			MinRunningInstances: application.MinRunningInstances,
-			MaxRunningInstances: application.MaxRunningInstances,
-			Healthchecks:        application.Healthchecks,
-			AutoPreview:         application.AutoPreview,
-			Ports:               application.Ports,
-			Arguments:           application.Arguments,
-			Entrypoint:          application.Entrypoint,
-			AutoDeploy:          *qovery.NewNullableBool(application.AutoDeploy),
-		}
-
-		if applicationBranch != "" {
-			req.GitRepository.Branch = &applicationBranch
-		}
-
+		var changeAutoDeploy = false
 		if cmd.Flags().Changed("auto-deploy") {
-			req.AutoDeploy = *qovery.NewNullableBool(&applicationAutoDeploy)
+			changeAutoDeploy = true
```

Comment:
iiuc you should target `&applicationAutoDeploy` instead of setting to `true`.
The condition `cmd.Flags().Changed("auto-deploy")` means that the flag has been set (it can be `true` or `false`).
Also, you should use a nullable variable type to be able to keep the current behavior, see https://github.com/Qovery/qovery-cli/pull/379/files#diff-347ce49b39961c56097bf8e9b5eee84236e37636a3615f43ab6197e9293bba7eR70